core/src/main/java/org/apache/carbondata/core/datamap/DataMapUtil.java

```diff
@@ -168,23 +168,32 @@ public static void pruneSegments(List<Segment> segments, List<ExtendedBlocklet>
 
   static List<ExtendedBlocklet> pruneDataMaps(CarbonTable table,
       FilterResolverIntf filterResolverIntf, List<Segment> segmentsToLoad,
-      List<PartitionSpec> partitions, List<ExtendedBlocklet> blocklets) throws IOException {
+      List<PartitionSpec> partitions, List<ExtendedBlocklet> blocklets,
+      DataMapChooser dataMapChooser) throws IOException {
+    if (null == dataMapChooser) {
+      return blocklets;
+    }
     pruneSegments(segmentsToLoad, blocklets);
     List<ExtendedBlocklet> cgDataMaps = pruneDataMaps(table, filterResolverIntf, segmentsToLoad,
         partitions, blocklets,
-        DataMapLevel.CG);
+        DataMapLevel.CG, dataMapChooser);
     pruneSegments(segmentsToLoad, cgDataMaps);
     return pruneDataMaps(table, filterResolverIntf, segmentsToLoad,
         partitions, cgDataMaps,
-        DataMapLevel.FG);
+        DataMapLevel.FG, dataMapChooser);
   }
 
   static List<ExtendedBlocklet> pruneDataMaps(CarbonTable table,
       FilterResolverIntf filterResolverIntf, List<Segment> segmentsToLoad,
-      List<PartitionSpec> partitions, List<ExtendedBlocklet> blocklets, DataMapLevel dataMapLevel)
+      List<PartitionSpec> partitions, List<ExtendedBlocklet> blocklets, DataMapLevel dataMapLevel,
+      DataMapChooser dataMapChooser)
       throws IOException {
-    DataMapExprWrapper dataMapExprWrapper =
-        new DataMapChooser(table).chooseDataMap(dataMapLevel, filterResolverIntf);
+    DataMapExprWrapper dataMapExprWrapper = null;
```

Comment:
Please call 'chooseDataMap' instead of if else

Reply:
fixed